src/overlays/ovl_i1/sf_tr360.c

```diff
@@ -1,9 +1,233 @@
-#include "common.h"
+#include "global.h"
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_tr360/func_i1_80198C50.s")
+extern ObjectInit D_6008EF8[];
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_tr360/func_i1_80198DCC.s")
+void func_i1_80198C50(void) {
+    Object_58* obj58;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_tr360/func_i1_80199024.s")
+    D_80178310 = SEGMENTED_TO_VIRTUAL(D_6008EF8);
+    Rand_SetSeed(1, 29000, 9876);
 
-#pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i1/sf_tr360/func_i1_8019949C.s")
+    obj58 = gObjects58;
+    for (i = 0; i < 1000; i++) {
+        if (D_80178310[i].id >= 0) {
+            if (D_80178310[i].id <= 160) {
+                func_800A4F4C(obj58);
+                obj58->obj.status = 2;
+                obj58->obj.id = D_80178310[i].id;
+                obj58->obj.pos.x = D_80178310[i].xPos;
+                obj58->obj.pos.z = -D_80178310[i].zPos1;
+                obj58->obj.pos.y = D_80178310[i].yPos - (Rand_ZeroOneSeeded() * 300.0f);
+                obj58->obj.rot.y = D_80178310[i].rot.y;
+                Object_SetInfo(&obj58->info, obj58->obj.id);
+                obj58++;
+            }
+        } else {
+            break;
+        }
+    }
+}
+
+bool func_i1_80198DCC(Actor* actor, f32 x, f32 z) {
+    Object_58* obj58;
+    s32 i;
+
+    for (i = 0, obj58 = gObjects58; i < 200;) {
+        if ((obj58->obj.status == 2) &&
+            (fabsf(obj58->obj.pos.x - (actor->obj.pos.x + ((x * 650.0f) + 350.0f))) < 1200.0f) &&
+            (fabsf(obj58->obj.pos.z - (actor->obj.pos.z + ((z * 650.0f) + 350.0f))) < 1200.0f) &&
+            (actor->obj.pos.y < 650.0f)) {
+            return TRUE;
```

Comment:
```suggestion
            return true;
```
We prefer the lowercase macros, even though we do have both. Also I still can't believe this isn't a loop unroll.